### PR TITLE
Remove PSStyle if terminal does not support escape sequence

### PR DESCRIPTION
**Pull request by lijinpei2008**
- Fix issue: https://github.com/Azure/azure-powershell/issues/24556

- Group in Azure-PowerShell PR: https://github.com/Azure/azure-powershell/pull/25735

- Current effect
image

- Changed effect
image

### CODE REVIEW

File: src/Common/AzurePSCmdlet.cs [hunk 1]
```diff
@@ -334,11 +334,21 @@ protected virtual void TearDownHttpClientPipeline()
             AzureSession.Instance.ClientFactory.RemoveUserAgent(ModuleName);
             AzureSession.Instance.ClientFactory.RemoveHandler(typeof(CmdletInfoHandler));
         }
+
+        protected virtual void InitializePSStyle()
+        {
+            if (null == PSStyle.SupportsVirtualTerminal)
+            {
+                PSStyle.SupportsVirtualTerminal = Host.UI.SupportsVirtualTerminal;
+            }
```
Comment (isra-fel) [inline src/Common/AzurePSCmdlet.cs:343]: This logic is too specific to be put in AzurePSCmdlet. It should be in PSStyle.

File: src/Common/ColorAndFormat/PSStyle.cs [hunk 2]
```diff
@@ -20,6 +20,8 @@ namespace Microsoft.WindowsAzure.Commands.Common
     /// </summary>
     public sealed class PSStyle
     {
+        public static bool? SupportsVirtualTerminal = null;
+
         /// <summary>
```
Comment (isra-fel) [inline src/Common/ColorAndFormat/PSStyle.cs:25]: Try something like this

```csharp
public static bool IsEscapeSequenceSupported {
  if (!_isEscapeSequenceSupported.HasValue) { _isEscapeSequenceSupported = Host.UI.SupportsVirtualTerminal; }
  return _isEscapeSequenceSupported.Value;
}
private static bool? _isEscapeSequenceSupported = null;
```

File: src/Common/AzurePSCmdlet.cs (hunk 1 above)
Comment (BethanyZhou) [reply src/Common/AzurePSCmdlet.cs:343]: Host.UI.SupportsVirtualTerminal is a property of PSCmdlet, its value is unavailable beyond cmdlet.

File: src/Common/ColorAndFormat/PSStyle.cs (hunk 2 above)
Comment (BethanyZhou) [reply src/Common/ColorAndFormat/PSStyle.cs:25]: Nice desgin

**Review by dolauli — COMMENTED** [review]
Do we need to add a change log somewhere for this change.

File: src/Common/ColorAndFormat/PSStyle.cs [hunk 3]
```diff
@@ -20,6 +22,29 @@ namespace Microsoft.WindowsAzure.Commands.Common
...
+                if (_isEscapeSequenceSupported.HasValue)
+                {
+                    return _isEscapeSequenceSupported.Value;
+                }
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// This function should be invoked only in the Az.Accounts.
```
Comment (isra-fel) [inline src/Common/ColorAndFormat/PSStyle.cs:40]:
```suggestion
        /// Initializes default settings of PSStyle according to the environment.
        /// Note: This function should be invoked only in Az.Accounts.
```

File: src/Common/ColorAndFormat/PSStyle.cs [hunk 4]
```diff
@@ -20,6 +22,29 @@ namespace Microsoft.WindowsAzure.Commands.Common
     /// </summary>
     public sealed class PSStyle
     {
+        private static bool? _isEscapeSequenceSupported = null;
+
+        public static bool IsEscapeSequenceSupported
```
Comment (isra-fel) [inline src/Common/ColorAndFormat/PSStyle.cs:27]: Any reason to make this public? Otherwise it can be private.

**Review by isra-fel — APPROVED** [review]

**Comment by BethanyZhou** [comment]
> Do we need to add a change log somewhere for this change.

@dolauli thanks for asking. No change log is needed for common repo.

**Review by dolauli — APPROVED** [review]

**Pull request merged**